Assets/MixedRealityToolkit.Services/InputAnimation/Editor/InputRecordingWindow.cs

```diff
@@ -0,0 +1,330 @@
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using UnityEngine;
+using UnityEngine.Playables;
+using UnityEngine.SceneManagement;
+using UnityEditor;
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Tools for recording and playing back input animation in the Unity editor.
+    /// </summary>
+    public class InputRecordingWindow : EditorWindow
+    {
+        private InputAnimation animation
+        {
+            get { return PlaybackService?.Animation; }
+            set { if (PlaybackService != null) PlaybackService.Animation = value; }
+        }
+
+        private string loadedFilePath = "";
+
+        private IMixedRealityInputRecordingService recordingService = null;
+        private IMixedRealityInputRecordingService RecordingService => recordingService ?? (recordingService = MixedRealityToolkit.Instance.GetService<IMixedRealityInputRecordingService>());
+
+        private IMixedRealityInputPlaybackService playbackService = null;
+        private IMixedRealityInputPlaybackService PlaybackService => playbackService ?? (playbackService = MixedRealityToolkit.Instance.GetService<IMixedRealityInputPlaybackService>());
+
+        private IInputSimulationService simService = null;
+        private IInputSimulationService SimService => simService ?? (simService = MixedRealityToolkit.Instance.GetService<IInputSimulationService>());
+
+        public enum ToolMode
+        {
+            /// <summary>
+            /// Record input animation and store in the asset.
+            /// </summary>
+            Recording,
+            /// <summary>
+            /// Play back input animation as simulated input.
+            /// </summary>
+            Playback,
+        }
+
+        private ToolMode mode = ToolMode.Recording;
+        public ToolMode Mode
+        {
+            get { return mode; }
+            private set
+            {
+                mode = value;
+            }
+        }
+
+        /// Icon textures
+        private Texture2D iconPlay = null;
+        private Texture2D iconPause = null;
+        private Texture2D iconRecord = null;
+        private Texture2D iconRecordActive = null;
+        private Texture2D iconStop = null;
+        private Texture2D iconStepFwd = null;
+        private Texture2D iconStepBack = null;
+        private Texture2D iconJumpFwd = null;
+        private Texture2D iconJumpBack = null;
+
+        [MenuItem("Mixed Reality Toolkit/Utilities/Input Recording")]
+        private static void ShowWindow()
+        {
+            InputRecordingWindow window = GetWindow<InputRecordingWindow>();
+            window.titleContent = new GUIContent("Input Recording");
+            window.minSize = new Vector2(380.0f, 680.0f);
+            window.Show();
+        }
+
+        private void OnGUI()
+        {
+            LoadIcons();
+
+            string[] modeStrings = { "Record", "Playback" };
```

Comment:
Super nit: Consider bringing this closer to the definition of the mode enum, just so that these two related and similarly-ordered entities are in the same rough area.